README.md

```diff
@@ -107,17 +108,16 @@ headers:
     <td>Filters for declarations.<br><b>Default: all are included</b></td>
     <td><pre lang="yaml"><code>
 functions:
-  include: # Exclude is also available.
-    names: # Matches with exact name.
-      - someFuncName
-      - anotherName
-    matches: # Matches using regexp.
-      - prefix.*
-      - [a-z][a-zA-Z0-9]*
-  prefix: 'cx_' # Prefix added to all functions.
-  prefix-replacement: # Replaces a functions's prefix.
-    'clang_': ''
-    '_': 'C'</code></pre></td>
+  include: # 'exclude' is also available.
+    - [a-z][a-zA-Z0-9]* # Matches using regexp.
+    - prefix.* # '.' matches any character.
+    - someFuncName # Matches with exact name
+    - anotherName # full names have higher priority.
```

Comment:
Are regexes automatically anchored with `^` and `$` or should I do that myself? It might be wise to give an example of this...

Also, how do you tell a regex from a full name? `anotherName` is a valid regex.. I would suggest either doing:
```yaml
include:
  names:
   - fullMethodName
  patterns:
  - ^prefix.*$
```
like before or:
```yaml
include:
  - fullMethodName
  - regex: ^prefix.*$  # Using an inline object with a key 'regex' to make a regular expression
```
or simply only regexes:
```yaml
include:
  - ^fullMethodName$
  - ^prefix.*$
```

Note. if anchors `^$` are not always automatically inserted around the patterns, I would suggest your examples highlights this. Similarly, if they are inserted around the patterns, the documentation should probably highlight this.

Reply:
> Are regexes automatically anchored

Nope, Do you think this should be necessary? I have not had any trouble using regexp matching till now in any examples or tests, when do you think will this be necessary?

>how do you tell a regex from a full name

We have actually had a pretty long discussion on this, and the final conclusion was to keep no subkeys for include/exclude. We do a simple regexp match to check if a regexp string only consists of `_, a-z, A-Z, 0-9` and if it does, we treat it as a full name, which has performance benefits because we can match in O(1) using a HashSet.

I do not have an in-depth knowledge of RegularExpressions, also we'd like to keep the regexp patterns simple enough, and its possible for us because the C declaration names we match with themselves have a limited pattern set.

Reply:
Just tested it with anchors on the `libclang-example`
```yaml
  structs:
    include:
        - '^CX.*$'
  enums:
    include:
      - '^CXTypeKind$'
```

Using or not using `^` `$` doesn't affect the output

Reply:
> We have actually had a pretty long discussion on this, and the final conclusion was to keep no subkeys for include/exclude. We do a simple regexp match to check if a regexp string only consists of _, a-z, A-Z, 0-9 and if it does, we treat it as a full name, which has performance benefits because we can match in O(1) using a HashSet.

Is performance a concern... if not I would focus on keep the code simple rather than fast :see_no_evil: 

In any case, I see that for the patterns you have this makes sense!